### DISCUSSION TITLE
Database config overrides config.xml?

### DISCUSSION THREAD

**Discussion by EricdeWaal** [gerbera/gerbera#1268] [Q&A]
Hi,

The new web config interface is great!
If there is an entry in config.xml and one for the same setting in the database, which one has priority?
And can I safely delete a section from the config.xml file once I managed to the this into the database?
I noticed that for some reason, when a setting is in congig.xml, it's visible in the web config but does not get stored in the database if I click Save Config.
Is this intentional? What is the best way to move settings from config.xml to the database?
A lot of questions, i know ;<)

Eric

**Comment by KarlStraussberger** [comment]
Database always has priority.

In the interface you can see where the value comes from: If you are able to reset a value it's from database. You may always reset to config.xml values, but if you delete an entry from config it's completely gone.

Only new or changed values are stored in the database. You cannot run gerbera without config file, but you may rebuild part of your config with the ui - which is not really intended. The intension is to change values without access to the server and without restarting the server (for some values at least).

**Comment by EricdeWaal** [comment]
Hi,

Right now only modified settings are stored in the database.
Is it a good idea to import all settings into the database, and eventually make the config file obsolete, except for the link to the database, like SQLite / MYSQL, location, dbname, username, password?
It will be easier for users not so familiar with Gerbera .
